extensions/iceberg/src/main/java/io/deephaven/iceberg/util/BuildCatalogOptions.java

```diff
@@ -30,11 +31,38 @@ public String name() {
     }
 
     /**
-     * The catalog properties. Must contain {@value CatalogUtil#ICEBERG_CATALOG_TYPE} or
+     * The catalog properties provided by the user. Must contain {@value CatalogUtil#ICEBERG_CATALOG_TYPE} or
      * {@value CatalogProperties#CATALOG_IMPL}.
+     * <p>
+     * For building the catalog, use {@link #updatedProperties()} instead.
      */
     public abstract Map<String, String> properties();
 
+    /**
+     * Whether Deephaven should not inject any additional properties (like for disabling CRT, setting a custom client
+     * credentials provider, etc.) into the properties map, if not already set by the user. This is enabled by default.
+     * <p>
+     * Disabling it would require users to set all the properties themselves.
+     */
+    @Value.Default
+    public boolean enablePropertyInjection() {
+        return true;
+    }
+
+    /**
+     * The properties to use when creating the catalog. This is a copy of the properties map, with any
+     * Deephaven-specific properties injected into it.
+     */
+    public Map<String, String> updatedProperties() {
+        if (!enablePropertyInjection()) {
+            return properties();
+        }
+        final Map<String, String> updatedProperties = new HashMap<>(properties());
+        // Inject Deephaven-specific AWS/S3 settings into the property map
+        AWSProperties.injectDeephavenProperties(updatedProperties);
+        return updatedProperties;
```

Comment:
Let's add some tests to `BuildCatalogOptionsTest` demonstrating the current behavior.